test/models/git/repository_test.rb

```diff
@@ -0,0 +1,69 @@
+require 'test_helper'
+
+module Git
+  class RepositoryTest < Minitest::Test
+    def test_lookup_commit_for_good_commit
+      repository = Repository.new(
+        :csharp,
+        repo_url: TestHelpers.git_repo_url("v3-monorepo")
+      )
+      assert repository.lookup_commit("88bfc517efcabd63714ee3b1d853d9bf233f4f3b")
+    end
+
+    def test_lookup_commit_for_bad_commit
+      repository = Repository.new(
+        :csharp,
+        repo_url: TestHelpers.git_repo_url("v3-monorepo")
+      )
+      assert_raises do
+        repository.lookup_commit("foobar")
+      end
+    end
+
+    def test_lookup_commit_for_head
+      repository = Repository.new(
+        :csharp,
+        repo_url: TestHelpers.git_repo_url("v3-monorepo")
+      )
+      assert_equal repository.head_commit, repository.lookup_commit("HEAD")
+    end
+
+    def test_read_json_blob_for_valid_path
```

Comment:
I've added tests for the existing `read_json_blob` and the new `read_json_text` method.

Reply:
Nice work! :)